src/shared/components/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+
+import { Modal } from './Modal';
+
+const mockOnClose = jest.fn();
+const mockOnSubmit = jest.fn();
+const mockOnSecondBtnSubmit = jest.fn();
+const mockOnThirdBtnSubmit = jest.fn();
+
+const commonProps = {
+  'data-testid': 'modal-test-id',
+  onClose: mockOnClose,
+  open: true,
+  title: 'Modal Title',
+  buttonText: 'Submit',
+  children: <div>Modal Content</div>,
+  onSubmit: mockOnSubmit,
+  onSecondBtnSubmit: mockOnSecondBtnSubmit,
+  onThirdBtnSubmit: mockOnThirdBtnSubmit,
+  secondBtnText: 'secondBtnText',
+  thirdBtnText: 'thirdBtnText',
+  hasSecondBtn: true,
+  hasThirdBtn: true,
+};
+
+describe('Modal Component', () => {
+  test('should render the modal with all elements', () => {
+    render(<Modal {...commonProps} />);
+
+    expect(screen.getByText(commonProps.title)).toBeInTheDocument();
```

Comment:
texts.forEach(text => expect(screen.getByText(text)).toBeInTheDocument());

you could do smth like that